pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -117,6 +118,15 @@ message CommandConnect {
 	optional string auth_method_name = 5;
 	optional bytes auth_data = 3;
 	optional int32 protocol_version = 4 [default = 0];
+
+	// Client can ask to be proxyied to a specific broker
+	// This is only honored by a Pulsar proxy
+	optional string proxy_to_broker_url = 6;
+
+	// Original principal that was verified by
+	// a Pulsar proxy. In this case the auth info above
+	// will the the auth of the proxy itself
+	optional string original_principal = 7;
```

Comment:
Proxy is not doing authorization on the data connections. The reason is that after the initial connect/connected it just forward everything on both sides.

The broker still needs to check whether that particular role is authorized to pulblish on a specific topic. So the proxy relays the original client principal, and that will be used by the broker for the authorization, trusting that the proxy has already validated the authentication.